### PR TITLE
Fix for not able to pass in default options on plugin register.

### PR DESCRIPTION
**Pull request by niekfrontmen**
Example with adding options to be passed in on registration which can be overridden on runtime.
  fastify.register(require('fastify-jwt'), {
    secret: `${fastify.config.jwt.secret}`,
    options: {
      issuer: 'Some issuer',
      subject: 'Some subject',
      audience: 'Some audience'
    }
  })

### CODE REVIEW

**Comment by mcollina** [comment]
tests are failing btw.

**Comment by niekheezemans** [comment]
Have a fix ready, will try to write tests asap

File: jwt.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ const {
   BadRequest,
   Unauthorized
 } = require('http-errors')
+let secretKey, secretPass
```
Comment (mcollina) [inline jwt.js]: those should not be defined at module level, but they should be wrapped in the plugin

File: package.json [hunk 2]
```diff
@@ -1,40 +1,74 @@
...
+  "main": "jwt.js",
+  "name": "fastify-jwt",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/fastify/fastify-jwt.git"
+  },
+  "scripts": {
+    "test": "standard && tap test.js"
+  },
+  "version": "0.4.0"
```
Comment (mcollina) [inline package.json]: Can you please restore this file?

File: jwt.js (hunk 1 above)
Comment (mcollina) [reply jwt.js]: ^ this still needs a fix.

File: test.js [hunk 3]
```diff
@@ -216,6 +253,128 @@ test('sign and verify', function (t) {
...
+    t.plan(2)
+
+    const fastify = Fastify()
+    fastify.register(jwt, { secret: 'test' })
+
+    fastify.post('/signSync', function (request, reply) {
+      reply.jwtSign(request.body)
+        .then(function (token) {
+          return reply.send({ token })
+        })
```
Comment (mcollina) [inline test.js]: Can you change this to:

```js
     return reply.jwtSign(request.body)
        .then(function (token) {
          return { token }
        })
```

In this way errors do not go to 'unhanledRejection'

File: test.js [hunk 4]
```diff
@@ -216,6 +253,128 @@ test('sign and verify', function (t) {
...
+    })
+
+    fastify.get('/verifySync', function (request, reply) {
+      request.jwtVerify()
+        .then(function (decodedToken) {
+          return reply.send(decodedToken)
+        })
```
Comment (mcollina) [inline test.js]: this should be  `return request.jwtVerify()`.

**Comment by mcollina** [comment]
Thanks!!

**Pull request merged**